tests/integration/ha_tests/application-charm/src/charm.py

```diff
@@ -88,17 +101,17 @@ def _count_writes(self) -> int:
         ) as connection, connection.cursor() as cursor:
             cursor.execute("SELECT COUNT(number) FROM continuous_writes;")
             count = cursor.fetchone()[0]
-        connection.close()
```

Comment:
With should be closing the connection anyway.

Reply:
I remembered about a note saying `leaving contexts doesn't close the connection` on https://www.psycopg.org/docs/connection.html#connection.